www/curj00/semestralka2/js/main.js

```diff
@@ -0,0 +1,372 @@
+    id: 40,
+    coords: "50°30'55.0\"N, 14°08'12.9\"E"
+}, {
+    name: "Olfin Car Trutnov",
+    id: 41,
+    coords: "50°33'12.0\"N, 15°54'15.4\"E"
+}, {
+    name: "Louda Auto Teplice",
+    id: 42,
+    coords: "50°39'44.5\"N, 13°52'26.0\"E"
+}, {
+    name: "Gerhard Horejsek Česká Lípa",
+    id: 43,
+    coords: "50°40'41.7\"N, 14°32'39.4\"E"
+}, {
+    name: "Gerhard Horejsek Děčín",
+    id: 44,
+    coords: "50°46'35.5\"N, 14°13'04.2\"E"
+}, {
+    name: "Manfred Schoner autoservis",
+    id: 45,
+    coords: "50°5'10.807\"N, 12°24'6.505\"E"
+}, {
+    name: "Louda Auto Praha",
+    id: 46,
+    coords: "50°6'29.972\"N, 14°32'54.237\"E"
+}, {
+    name: "Autotrend",
+    id: 47,
+    coords: "50°44'36.438\"N, 15°2'53.052\"E"
+}];
+
+var znacky = [];
+var souradnice = [];
+var startovniBod;
+
+data.forEach(function (marker) { /* Vyrobit markery */
+    var c = SMap.Coords.fromWGS84(marker.coords); /* Souřadnice značky, z textového formátu souřadnic fromWGS84(14.297847, 50.076322);*/
+    var options = {
+        url: obrazek,
+        title: marker.name,
+        anchor: { left: 10, bottom: 1 }  /* Ukotvení značky za bod uprostřed dole */
+    }
+    // Důležité je přiřazení ID jednotlivým markerům - vlastní ID, jinak se generuje nahodne
+    var znacka = new SMap.Marker(c, marker.id, options);
+    souradnice.push(c);
+    znacky.push(znacka);
+});
+
+/* NH Car ukotvíme za střed značky, přestože neznáme její velikost */
+// var options = {
+//     anchor: { left: 0.5, top: 0.5 }
+// }
+// znacky[1].decorate(SMap.Marker.Feature.RelativeAnchor, options);
+
+
+var vrstva = new SMap.Layer.Marker();     /* Vrstva se značkami */
+mapa.addLayer(vrstva);                          /* Přidat ji do mapy */
+vrstva.enable();                         /* A povolit */
+for (var i = 0; i < znacky.length; i++) {
+    vrstva.addMarker(znacky[i]);
+    var card = new SMap.Card();
+    card.getBody().innerHTML = "<br>Prodejce vozů Volkswagen";
+    card.setSize(270, 90);
+    card.getHeader().innerHTML = "<strong>" + data[i].name + "</strong>";
+    znacky[i].decorate(SMap.Marker.Feature.Card, card);
+}
+
+
+
+var cz = mapa.computeCenterZoom(souradnice); /* Spočítat pozici mapy tak, aby značky byly vidět */
+//mapa.setCenterZoom(cz[0], cz[1]);
+var trasa;
+var vrstvaSTrasou = new SMap.Layer.Geometry();
+mapa.addLayer(vrstvaSTrasou).enable();
+
+var nalezeno = function (route) {
+    var coords = route.getResults().geometry;
+    var cz = mapa.computeCenterZoom(coords);
+    //mapa.setCenterZoom(cz[0], cz[1]);
+    if (trasa != null) {
+        vrstvaSTrasou.removeGeometry(trasa);
+    }
+    trasa = new SMap.Geometry(SMap.GEOMETRY_POLYLINE, null, coords);
+    vrstvaSTrasou.addGeometry(trasa);
+}
+
+var markerClicked = false;
+var endPointCoords;
+
+function handleMarkerClick(e) { /* Kliknutí na marker */
+    markerClicked = true;
+    // Vybraný marker
+    var marker = e.target;
+    var id = marker.getId();
+    loadDoc(id);
```

Comment:
proc tady nacitas vzdycky to same a jeste k tomu uplne vsechny body, ktere tady vubec nepotrebujes ?